### ISSUE TITLE
Page breaking before <form>

### ISSUE THREAD

**Issue by rgmrtn** [Kozea/WeasyPrint#1249]
The weasyprint PDF output is breaking the page just before a form element, when the form element is bigger than a page. I've tried adding the following style to the page:

```css
*{
  break-inside: avoid !important;
  page-break-inside: avoid !important;
}
form{
  break-before: avoid !important;
  page-break-before: avoid !important;
  break-inside: auto !important;
  page-break-inside: auto !important;
}
```

to prevent page breaks anywhere but in the form, but it still breaks the page before the `<form>` element, and not in the form element.

Any advice?

**Comment by liZe** [comment]
Hi!

There’s nothing special in forms about page breaks, it should work. Could you please provide a sample or an example showing your problem?

**Comment by rgmrtn** [comment]
Hi @liZe 

Here is the HTML code https://hastebin.com/ilicaxirip.http , and attached is the PDF with the issue.

Thank you
[AGL-20201112-Near_Miss_Form_sub59.pdf](https://github.com/Kozea/WeasyPrint/files/5548196/AGL-20201112-Near_Miss_Form_sub59.pdf)

**Comment by liZe** [comment]
The problem is caused by #36: there’s currently no page break in flex boxes, that are set on `.row` elements by Bootstrap.

A simple workaround for your use case is to render these rows as blocks, but it will probably break other things (columns at least):

```css
.row {
  display: block;
}
```

Unfortunately, there’s no simple solution until we fix #36.

**Comment by rgmrtn** [comment]
Thank you this did the trick!
